src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSMessagingUtils.cs

```diff
@@ -92,7 +92,7 @@ internal static PropagationContext ExtractParentContext(SNSEvent snsEvent)
         return ExtractParentContext(record);
     }
 
-    internal static PropagationContext ExtractParentContext(SNSEvent.SNSRecord record)
+    internal static PropagationContext ExtractParentContext(SNSEvent.SNSRecord? record)
```

Comment:
We have a null-check in place in at least one call. Can we make this parameter non-null & remove the inner null check?

Reply:
I wouldn't do it this method is called from `ExtractParentContext(SNSEvent snsEvent)` with the argument which can potentially be null: `snsEvent?.Records?.LastOrDefault()`

Reply:
In that case, can we instead remove the null check before the other call?

Reply:
Yes, we can. Removed.